### DISCUSSION TITLE
Unable to install Tailwind CSS v3.1.8 using Next.js framework guide

### DISCUSSION THREAD

**Discussion by ShashwatNautiyal** [tailwindlabs/tailwindcss#9158] [Help]
Tailwind: v3.1.8
Next.js v12.2.5
Node.js: v18.5.0
Browser: Chrome
OS: macOS

I'm not able to install tailwind CSS by running the below command which is given in Tailwind's Next.js framework guide.

`npm install -D tailwindcss postcss autoprefixer`

I tried running these commands as per the tailwind framework docs but there seems to be a problem and the error is thrown on terminal `npm ERR! Cannot read properties of null (reading 'matches')`

<img width="759" alt="image" src="https://user-images.githubusercontent.com/85606496/186009844-02e633f5-3fdd-419c-814c-955eaab27e23.png">

**Comment by thecrypticace** [comment]
This works fine for me but I've noticed that create-next-app has started using Yarn if you have it installed. What version of Yarn do you have installed? I'm thinking this could be related.

**Comment by ShashwatNautiyal** [comment]
Next.js documentation has mentioned that we can use `create-next-app` with 3 package managers and the next.js is working fine, problem is with tailwind command. 

<img width="677" alt="image" src="https://user-images.githubusercontent.com/85606496/186073823-efccf476-9e86-4049-a023-bef1ace8b9f8.png">

And I think the problem is with one of the libraries from `tailwindcss postcss autoprefixer` because the next.js is working fine with npm, yarn and pnpm. There might be an issue in postcss or autoprefixer which has broken the `npm install -D tailwindcss postcss autoprefixer` command. 

I have tried using the yarn package manager and it is working fine.

I'm using the yarn version - 1.22.19

**Comment by Anas-Alhariri** [comment]
After trying and reinstalling nodejs, nothing worked.

I figured out from the beginning that nextjs was using pnpm behind the seen, as the package-lock.json wasn't added after creating the next app. But instead, the pnpm-lock.yaml was created.

So, according to the @ShashwatNautiyal  yarn worked. I suspected that yarn is being used behind the seen just like pnpm. So, I installed yarn
``` npm install -g yarn```

then ran the normal command again:

```npm install -D tailwindcss postcss autoprefixer```

and everything just worked!

I hope you find this helpful.
